### DISCUSSION TITLE
[mini.ai] visually select jsx html tags

### DISCUSSION THREAD

**Discussion by ShooTeX** [nvim-mini/mini.nvim#285] [Q&A] [answered]
`vat` doesn't work correctly in some cases, sorry if this is a duplicate, but I didn't quite get what I have to do to get this working. The video shows the, _for me_, weird behavior

- first try: the selection is what I expected
- secondy try: the parent element is getting selected, not what I was expecting

https://user-images.githubusercontent.com/10850738/228384862-f21dc219-e55c-457c-b4d5-366e68df74a7.mov

I'm using LazyVim, and I started the first discussion there
https://github.com/LazyVim/LazyVim/discussions/500

**Comment by echasnovski** [comment, marked as answer]
The reason seems to be that it is a code with nested same tags. It is the most visible condition at which Lua pattern based approach is not good. See [this comment](https://github.com/echasnovski/mini.nvim/issues/156#issuecomment-1308429031) for more information. And also [this other comment](https://github.com/echasnovski/mini.nvim/issues/110#issuecomment-1212277863) with a more proper tree-sitter based specification.
